substrate/frame/nomination-pools/src/migration.rs

```diff
@@ -22,56 +22,246 @@ use sp_std::{collections::btree_map::BTreeMap, vec::Vec};
 
 #[cfg(feature = "try-runtime")]
 use sp_runtime::TryRuntimeError;
+pub mod v6 {
+	use super::*;
 
-pub mod v1 {
+	/// Wrapper over `MigrateToV6` with convenience version checks.
+	pub type VersionedMigrateV6<T> = frame_support::migrations::VersionedMigration<
+		5,
+		6,
+		MigrateToV6<T>,
+		crate::pallet::Pallet<T>,
+		<T as frame_system::Config>::DbWeight,
+	>;
+
+	/// This migration would restrict reward account of pools to go below ED by doing a named
+	/// freeze on all the existing pools.
+	pub struct MigrateToV6<T>(sp_std::marker::PhantomData<T>);
```

Comment:
I would not even expose this as `pub`.

Reply:
Unfortunately it needs to be, because it's used in a pub struct.

The best you can do (and I'd like to become a new standard) is wrapping the unversioned upgrade in a private module.

See details here: https://github.com/paritytech/polkadot-sdk/blob/liam-migrations-reference-docs/substrate/frame/examples/storage-migrations/src/migrations/v1.rs#L39-L54